payara/monitoring-console
# Payara InSight Enterprise UI/UX

### Summary
Complete UI/UX overhaul of Mon.. InSight. 

What can I say: most things UI/UX did change. Except the graphs themselves stayed pretty much what they were.

On the Java/REST API side of things there is only a tiny change adding a new field to the watch state making it possible to show how long the watch has been in its current state.

All other changes are client side and only affect the rendering and behaviour of the application in the browser.

A summary:

* page got a copyright footer
* page got a favicon
* app includes and uses custom font
* app includes and uses SVG icons and new logo SVGs
* navigation and quick settings in left sidebar
* settings in right sidebar now utilise tabs and got reorganised as with tabs you only can look at a single group at a time
* buttons to add new pages, widgets, watches are placed and marked more prominently
* increased use of modal dialogs
* destructive actions use a confirmation dialog (shown red if potentially causes a server state change)
* modifying actions show success or error messages
* some settings checkboxes are now rendered as toggle switches (where it made sense)
* query based pages can now use a widget type filter (was needed to produce a "clean" RAG page)
* RAG page is now called _Traffic Light Status_ as _RAG_ wasn't the common term I took it to be
* ... lots of styling details

### Testing
1. build this PR
2. in order to use InSight from this PR change the root `pom.xml` of Payara Enterprise accordingly:
```xml
-        <monitoring-console-api.version>1.1</monitoring-console-api.version>
-        <monitoring-console-process.version>1.2.1</monitoring-console-process.version>
-        <monitoring-console-webapp.version>1.2.1</monitoring-console-webapp.version>
+        <monitoring-console-api.version>1.2-SNAPSHOT</monitoring-console-api.version>
+        <monitoring-console-process.version>1.4-SNAPSHOT</monitoring-console-process.version>
+        <monitoring-console-webapp.version>1.4-SNAPSHOT</monitoring-console-webapp.version>
```
3. build and deploy Payara Enterprise with changed pom
4. start it using `asadmin start-domain production`
5. open admin-console, click link _Open Monitoring Console_ on the dashboard

Now you should see the new InSight (might take a browser refresh if you have been testing MC before).
Also you might want to delete local storage for the page to get rid of any old state.

From here just go crazy and try to click on everything ;) 
It might be a good idea to do so as _User_ and later also test as _Administrator_ (Role switched in app settings).
